ddtrace/tracer/textmap.go

```diff
@@ -831,21 +828,16 @@ func parseTraceparent(ctx *spanContext, header string) error {
 	if len(fullTraceID) != 32 {
 		return ErrSpanContextCorrupted
 	}
-	// checking that the entire TraceID is a valid hex string
 	if ok, err := regexp.MatchString("^[a-f0-9]+$", fullTraceID); !ok || err != nil {
 		return ErrSpanContextCorrupted
 	}
-	var err error
-	if ctx.traceID, err = strconv.ParseUint(fullTraceID[16:], 16, 64); err != nil {
-		return ErrSpanContextCorrupted
+	if ctx.trace != nil {
+		// Ensure that the 128-bit trace id tag doesn't propagate
```

Comment:
Is this done so that the `extractTraceID128` can "properly" set the full 128 bit id?